write.go

```diff
@@ -53,14 +52,14 @@ type msgWriter struct {
 
 func (mw *msgWriter) Write(p []byte) (int, error) {
 	if mw.closed {
-		return 0, errors.New("cannot use closed writer")
+		return 0, errClosed
```

Comment:
These two are separate errors and should remain. They aren't the same as `net.ErrClosed`.